atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidator.java

```diff
@@ -237,47 +240,51 @@ private Set<CassandraServer> getNewHostsWithInconsistentTopologiesFromTopologyRe
                 // of what the old servers were thinking, since in containerised deployments all nodes can change
                 // between refreshes for legitimate reasons (but they should still refer to the same underlying
                 // cluster).
-                if (pastConsistentTopologies.get() == null) {
+                ConsistentClusterTopologies pastTopologies = pastConsistentTopologies.get();
+                if (pastTopologies == null) {
                     // We don't have a record of what worked in the past, and since this state means we're validating
                     // the initial config servers, we don't have another source of truth here.
                     return newServersWithoutSoftFailures.keySet();
                 }
-                Optional<ConsistentClusterTopologies> maybeTopologies = maybeGetConsistentClusterTopology(
+                Map<CassandraServer, IdSupportingHostIdResult> matchingServers = EntryStream.of(
                                 serversToConsiderWhenNoQuorumPresent)
-                        .agreedTopologies();
-                if (maybeTopologies.isEmpty()) {
-                    log.info(
-                            "No quorum was detected in original set of servers, and the filtered set of servers were"
-                                    + " also not in agreement. Not adding new servers in this case.",
-                            SafeArg.of("newServers", CassandraLogHelper.collectionOfHosts(newlyAddedHosts)),
-                            SafeArg.of("allServers", CassandraLogHelper.collectionOfHosts(allHosts)));
-                    return newServersWithoutSoftFailures.keySet();
-                }
-                ConsistentClusterTopologies newNodesAgreedTopologies = maybeTopologies.get();
+                        .filterValues(result -> result.result().type() == HostIdResult.Type.SUCCESS
```

Comment:
yeah let's do this! I thought about it - was a bit hesitant to include the Type field because it was on HostIdResult, but these are pretty coupled anyway